### PR TITLE
2839: search view updates

### PR DESCRIPTION
**Pull request by rachaelparris**
Screen Shot 2019-11-26 at 12 56 47 PM

### CODE REVIEW

File: web-client/src/presenter/actions/AdvancedSearch/setDocketNumberFromAdvancedSearchAction.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * @returns {object} the caseId set to the docketNumber provided in the search term
+ */
+export const setDocketNumberFromAdvancedSearchAction = ({ get, store }) => {
+  const searchTerm = get(state.advancedSearchForm.docketNumber);
+  const docketNumber = trimDocketNumberSearch(searchTerm);
+  //TODO - refactor this. why are we setting state.caseId to docketNumber?
+  //also see setCaseIdFromSearchAction and navigateToCaseDetailAction
+  store.set(state.caseId, docketNumber);
+  return {
+    caseId: docketNumber,
```
Comment (matthopson) [inline web-client/src/presenter/actions/AdvancedSearch/setDocketNumberFromAdvancedSearchAction.js:19]: `caseId` and `docketNumber` are different things, right?

Comment (matthopson) [reply web-client/src/presenter/actions/AdvancedSearch/setDocketNumberFromAdvancedSearchAction.js:19]: ...which is literally in the comment above that line....

**Review by matthopson — APPROVED** [review]

**Review by codyseibert — APPROVED** [review]

File: web-client/src/applicationContextPublic.js [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
   getCognitoLoginUrl,
   getPublicSiteUrl,
 } from '../../shared/src/sharedAppContext.js';
+import { validateCaseAdvancedSearchInteractor } from '../../shared/src/business/useCases/validateCaseAdvancedSearchInteractor';
+import axios from 'axios';
+import deepFreeze from 'deep-freeze';
 
 const applicationContextPublic = {
+  getBaseUrl: () => {
+    return process.env.API_URL || 'http://localhost:3000';
```
Comment (kkoskelin) [inline web-client/src/applicationContextPublic.js:12]: For `public`, isn't it port `5678`? Or maybe I'm mixed up.

Comment (rachaelparris) [reply web-client/src/applicationContextPublic.js:12]: This is the URL for the API, which they'll be sharing.

File: web-client/src/presenter/actions/AdvancedSearch/setDocketNumberFromAdvancedSearchAction.js (hunk 1 above)
Comment (rachaelparris) [reply web-client/src/presenter/actions/AdvancedSearch/setDocketNumberFromAdvancedSearchAction.js:19]: Going to clean up (in this and other places) in the next PR.

File: web-client/src/views/AdvancedSearch/SearchForm.jsx [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+            <div className="header-with-blue-background display-flex flex-justify">
+              <h3>Search by Name</h3>
+              <Button
+                link
+                className="margin-left-1 tablet:margin-left-205 margin-right-0 padding-0 ustc-button--mobile-inline"
+                onClick={() => {
+                  clearAdvancedSearchFormSequence();
+                }}
+              >
+                <FontAwesomeIcon icon={['fas', 'times-circle']} />
```
Comment (kkoskelin) [inline web-client/src/views/AdvancedSearch/SearchForm.jsx]: I think the `Button` component can take an `icon` parameter for this now.

**Review by ericsorenson — APPROVED** [review]

File: web-client/src/views/AdvancedSearch/SearchForm.jsx [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+                      <fieldset className="usa-fieldset margin-bottom-0">
+                        <legend
+                          className="display-block"
+                          id="year-filed-legend"
+                        >
+                          Year filed
+                        </legend>
+                        <div className="usa-form-group--year display-inline-block">
+                          <input
+                            aria-describedby="year-filed-label"
```
Comment (kkoskelin) [inline web-client/src/views/AdvancedSearch/SearchForm.jsx]: There are two references here to `year-filed-label` but I don't see an element with that ID.

File: web-client/src/views/AdvancedSearch/SearchForm.jsx [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+            <div className="header-with-blue-background display-flex flex-justify">
+              <h3>Search by Docket Number</h3>
+              <Button
+                link
+                className="margin-left-1 tablet:margin-left-205 margin-right-0 padding-0 ustc-button--mobile-inline"
+                onClick={() => {
+                  clearAdvancedSearchFormSequence();
+                }}
+              >
+                <FontAwesomeIcon icon={['fas', 'times-circle']} />
```
Comment (kkoskelin) [inline web-client/src/views/AdvancedSearch/SearchForm.jsx]: same `<Button icon="..." />`

**Review by kkoskelin — CHANGES_REQUESTED** [review]
A few things I think probably could be tweaked, but I welcome corrections.

**Review by kkoskelin — APPROVED** [review]

**Pull request merged**